src/worker/upstreamcommands.js

```diff
@@ -538,6 +531,40 @@ commands['366'] = async function(msg, con) {
     return false;
 };
 
+commands.MODE = async function(msg, con) {
+    const raw_modes = msg.params[1];
+    const raw_params = msg.params.slice(2);
+    const parsedModes = parseMode(con, raw_modes, raw_params);
+
+    let buffer = con.state.getBuffer(msg.params[0]);
+
+    if (!buffer) {
+        return;
+    }
+
+    parsedModes.forEach((m) => {
+        if (m.param) {
+            return;
+        }
+        if (m.mode[0] === '+') {
+            buffer.modes += m.mode[1];
```

Comment:
This should probably be safer and make sure we don't already have the mode in `buffer.modes`